src/pages/iou/steps/IOUAmountPage.js

```diff
@@ -65,6 +58,45 @@ class IOUAmountPage extends React.Component {
         }
     }
 
+    /**
+     * Update amount with number or Backspace pressed.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit to enable Next button
+     *
+     * @param {String} key
+     * @param {String} event
+     */
+    onlyAllowNumericValues(key, event) {
+        // Prevent reusing synthetic event object, so we can call preventDefault() later.
+        if (event) {
+            event.persist();
```

Comment:
I'm not really following why this is necessary. Can we just call `event.preventDefault()` after we check for the backspace button and the new value and then not `setState()`?

Reply:
Actually, we don't need to call `event.preventDefault()` anymore as we started using value instead of default value.